synapse/cmds/cortex.py

```diff
@@ -0,0 +1,159 @@
+import synapse.lib.cli as s_cli
+import synapse.lib.tufo as s_tufo
+import synapse.lib.scope as s_scope
+
+class AskCmd(s_cli.Cmd):
+
+    _cmd_name = 'ask'
+    _cmd_syntax = (
+        ('--debug',{}),
+        ('query',{'type':'glob'}),
+    )
+
+    def runCmdOpts(self, opts):
+        ques = opts.get('query')
+        core = s_scope.get('syn:cmd:core')
+        if core == None:
+            self.printf('no connected cortex. see "open" cmd.')
+            return None
+
+        resp = core.ask(ques)
+        # {'oplog': [{'mnem': 'lift', 'add': 0, 'took': 1, 'sub': 0}], 'data': [], 'options': {'uniq': 1}, 'limits': {'touch': None, 'lift': None, 'time': None}}
+
+        if opts.get('debug'):
+
+            self.printf('oplog:')
+            for opfo in resp.get('oplog'):
+                mnem = opfo.get('mnem')
+                took = opfo.get('took')
+                self.printf('    %s (took:%d) %r' % (mnem,took,opfo))
+
+            self.printf('options:')
+            for name,valu in sorted(resp.get('options').items()):
+                self.printf('    %s = %s' % (name,valu))
+
+            self.printf('limits:')
+            for name,valu in sorted(resp.get('limits').items()):
+                self.printf('    %s = %s' % (name,valu))
+
+        def nodevalu(t):
+            return t[1].get( t[1].get('tufo:form') )
+
+        nodes = list(sorted( resp.get('data'), key=nodevalu))
+
+        for node in nodes:
+            form = node[1].get('tufo:form')
+            valu = node[1].get(form)
+
+            tags = s_tufo.tags(node)
+
+            # FIXME local typelib and datamodel
```

Comment:
This comment refers to the fact that calling `core.getPropRepr` will round trip each time.  In the future, we should pull down the remote data model and work against it locally.